packages/material-ui/src/styles/variants.d.ts

```diff
@@ -1,17 +1,9 @@
-import { CSSProperties, CreateCSSProperties, PropsFunc } from '@material-ui/styles/withStyles';
+import { CSSInterpolation } from './experimentalStyled';
 import { ComponentsPropsList } from './props';
 
 export type ComponentsVariants = {
   [Name in keyof ComponentsPropsList]?: Array<{
     props: Partial<ComponentsPropsList[Name]>;
-    style: // JSS property bag
-    | CSSProperties
-      // JSS property bag where values are based on props
-      | CreateCSSProperties<Partial<ComponentsPropsList[Name]>>
-      // JSS property bag based on props
-      | PropsFunc<
```

Comment:
As a side note (as the value here seems to be about sharing the same types), there might be something to consider regarding `PropsFunc`. It's currently not implemented but the type accepts it: https://codesandbox.io/s/globalthemeoverride-material-demo-forked-x099w?file=/demo.tsx. I would personally +1 for making it work. I'm not aware we have an issue open about it.

Reply:
I'm not aware we have an open issue about it, I would propose we open one.

Reply:
I'll look into this next week 👍

Reply:
I think that it depends on what we want to support for the API?

I would personally use the props & `styleProps` from the `styleOverrides` key. As for the `variants` key, unclear.

Reply:
Ok let's start the discussion on an dedicated PR 👍